src/base/bittorrent/session.cpp

```diff
@@ -3894,56 +3912,96 @@ void Session::startUpTorrents()
         ++resumedTorrentsCount;
     };
 
-    qDebug("Starting up torrents");
+    qDebug("Starting up torrents...");
     qDebug("Queue size: %d", fastresumes.size());
-    // Resume downloads
-    QMap<int, TorrentResumeData> queuedResumeData;
-    int nextQueuePosition = 1;
-    int numOfRemappedFiles = 0;
+
     const QRegularExpression rx(QLatin1String("^([A-Fa-f0-9]{40})\\.fastresume$"));
-    foreach (const QString &fastresumeName, fastresumes) {
-        const QRegularExpressionMatch rxMatch = rx.match(fastresumeName);
-        if (!rxMatch.hasMatch()) continue;
 
-        QString hash = rxMatch.captured(1);
-        QString fastresumePath = resumeDataDir.absoluteFilePath(fastresumeName);
-        QByteArray data;
-        CreateTorrentParams torrentParams;
-        MagnetUri magnetUri;
-        int queuePosition;
-        if (readFile(fastresumePath, data) && loadTorrentResumeData(data, torrentParams, queuePosition, magnetUri)) {
-            if (queuePosition <= nextQueuePosition) {
-                startupTorrent({ hash, magnetUri, torrentParams, data });
-
-                if (queuePosition == nextQueuePosition) {
-                    ++nextQueuePosition;
-                    while (queuedResumeData.contains(nextQueuePosition)) {
-                        startupTorrent(queuedResumeData.take(nextQueuePosition));
-                        ++nextQueuePosition;
+    if (isQueueingSystemEnabled()) {
+        QFile queueFile {resumeDataDir.absoluteFilePath(QLatin1String {"queue"})};
+
+        // TODO: Remove the deprecated code below at v4.2
```

Comment:
>// TODO: Remove the deprecated code below at v4.2

Isn't this way too soon? Shouldn't it be removed after some months inside 4.2.x releases or at 4.3.x release?

Reply:
I could delete it even now. It just will break the queue once, which is already prone to breaking due to the current way.
I left it to avoid angry comments from fans of (pseudo-)smooth updates.

Reply:
Doesn't this break the queue only if someone: Installs this version, then downgrades, runs it and changes queue numbers and finally re-upgrades. Then the new version will pickup a stale `queue` file.

Reply:
>Installs this version, then downgrades, runs it and changes queue numbers and finally re-upgrades. Then the new version will pickup a stale queue file.

In such scenario the queue will be broken anyway (having or not this "deprecated" code). And I don't think it can (should) be solved.

I meant the following:
1. If we keep "deprecated" code it loads queue positions (possible broken; see #9556) from .fastresume files at first run of updated qBittorrent. Then "queue" file will be used (unless user removes it manually). So it affected by the problem above.
2. If we remove "deprecated" code "queue" file will be used right away so it breaks queue positions at first run of updated qBittorrent (due to missing queue file). It also affected by the problem above.

Reply:
>And I don't think it can (should) be solved.

+1

If I understand correctly your 2 points are in case "the user upgrades from old system to new system. 1st point is with migration code and 2nd point is without it".

So if we just keep the migration code for a reasonable amount of releases we should eliminate most of the problem in the long run.
Sidenote: IMO, the other -older- migration paths should be removed together with this one. Those are a lot older anyway.

Reply:
>If I understand correctly your 2 points are in case "the user upgrades from old system to new system. 1st point is with migration code and 2nd point is without it".

Yes.
>So if we just keep the migration code for a reasonable amount of releases we should eliminate most of the problem in the long run.

Maybe. If once broken queue is a real problem.
Just let me know what I have to say in this comment. And what do you think about what it should be, just a TODO comment (like I did), or something suggested by other reviewers?
>the other -older- migration paths should be removed together with this one. Those are a lot older anyway.

+1

Reply:
>Maybe. If once broken queue is a real problem.

Then we could just leave the migration code for more releases in, like we have done with even older migration paths (which can be removed now IMO).

>And what do you think about what it should be, just a TODO comment (like I did), or something suggested by other reviewers?

Since we don't know beforehand when exactly we will remove this migration code, the `TODO` item is fine IMO.

Reply:
>Just let me know what I have to say in this comment.

Something like:
```c++
// TODO: Migration code introduced in 4.1.5. Remove after several releases in 4.2.x.
```

Reply:
>Migration code introduced in 4.1.5.

I don't introduce any migration code. I just do `if (queue file exists) use new startup logic; else use old one;`.
So I just use old code conditionally.

Reply:
I guess it is semantics then. For lack of (my) imagination:
> // TODO: Introduced in 4.1.5. Remove after several releases in 4.2.x.

Reply:
Sure. No problem.

Reply:
Done!